velox/functions/prestosql/Rand.h

```diff
@@ -56,4 +56,67 @@ struct RandFunction {
   }
 };
 
+template <typename T>
+struct SecureRandFunction {
+  static constexpr bool is_deterministic = false;
+
+  FOLLY_ALWAYS_INLINE void call(double& result) {
+    result = folly::Random::secureRandDouble01();
+  }
+
+  FOLLY_ALWAYS_INLINE void
+  call(double& out, const double lower, const double upper) {
+    VELOX_USER_CHECK_GE(lower, 0.0, "lower bound must be positive");
```

Comment:
I don't see the "lower bound must be positive" and "upper bound must be positive" errors in the java code. Why have you added these ?

Does Presto java not support such inputs ?

Would it be possible to do secureRandom(0, upper - lower) + lower to obtain a secure random value between lower and upper if non-negative values are not supported in the folly functions ?

Reply:
Thanks @aditi-pandit .

Removed the checks for lower and upper bounds to be positive.

I could not verify the java implementation of secure_random(). But rand() was not accepting negative values.
```
presto> select secure_random();
Query 20230917_072758_00009_cnqcw failed:  Scalar function name not registered: presto.default.secure_random, called with arguments: ().

presto> select secure_random(1,10);
Query 20230917_072811_00010_cnqcw failed:  Scalar function name not registered: presto.default.secure_random, called with arguments: (INTEGER, INTEGER).

presto> select rand();
       _col0        
--------------------
 0.3315594684084775 
(1 row)
presto> select rand(5);
 _col0 
-------
     2 
(1 row)
presto> select rand(-5);
Query 20230917_072729_00008_cnqcw failed: input > 0 (-5 vs. 0) bound must be positive presto.default.rand(-5:INTEGER)

presto> 
```